raptor/control_server.py

```diff
@@ -90,3 +90,4 @@ def start(self):
     def stop(self):
         LOG.info("shutting down control server")
         self.server.shutdown()
+        self._server_thread.join()
```

Comment:
Oh cool, so this blocks the current thread (stop()) until the actual server thread itself is terminated, sweet!